cmd/podman/images/list.go

```diff
@@ -128,10 +129,27 @@ func images(cmd *cobra.Command, args []string) error {
 		return err
 	}
 
-	imgs, err := sortImages(summaries)
+	reference := ""
+	for _, filter := range listOptions.Filter {
+		if strings.HasPrefix(filter, "reference=") {
+			reference = strings.TrimPrefix(filter, "reference=")
+			if strings.HasPrefix(reference, "sha256:") {
+				reference = ""
+				break
+			}
+			reference, _, _ = tokenRepoTag(reference)
+			if strings.Contains(reference, "<none>") {
```

Comment:
should this be an `==` not contains? But again it is not clear to me why this is done at all.

This whole section should have proper comments what is being filtered here.